pyfr/plugins/acoustics.py

```diff
@@ -0,0 +1,2106 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict, namedtuple
+from mpi4py import MPI
+import numpy as np
+import os
+import re
+import sys
+
+from pyfr.inifile import Inifile
+from pyfr.mpiutil import get_comm_rank_root, get_mpi
+from pyfr.nputil import CubicSplineFit, fuzzysort, LinearFit, npeval
+from pyfr.plugins.base import BasePlugin
+from pyfr.writers.native import NativeWriter
+
+
+def open_csv(fname, header=None, mode='w'):
+    # Open for mode
+    outf = open(fname, mode)
+    # Output a header if required
+    if os.path.getsize(fname) == 0 and header:
+        print(header, file=outf)
+    return outf
+
+def write_fftdata(fname, *indata, header='', mode='a'):
+    #open the file
+    outf = open_csv(fname, ''.join(header), mode)
+    #prepare the data array
+    data = np.array(indata)
+    dshape = data.shape
+    #do write
+    for i in range(dshape[1]):
+        print(*data[:, i].ravel(), sep=' ', file=outf)
+    outf.flush()
+    return
+
+def realfft(udata_):
+        dsize = udata_.shape[-1]
+        # freqsize: size of half the spectra with positive frequencies
+        freqsize = int(dsize/2) if dsize%2 == 0 else int(dsize-1)/2
+        ufft = np.fft.rfft(udata_)/freqsize 
+        return ufft
+
+def compute_spl(pref, amp, df=None):
+    sqrt2 = 1./np.sqrt(2.) 
+    lgsqrt2 = 20.*np.log10(sqrt2)
+    factor = sqrt2/pref if not df else sqrt2/(pref*np.sqrt(df))
+    spl = 20.*np.log10(factor*amp) 
+    spl[0] -= lgsqrt2
+    spl[-1] -= lgsqrt2
+    #overall sound pressure level
+    ospl_sum = sum(pow(10., spl/10.)) if not df else df*sum(pow(10., spl/10.))
```

Comment:
`10**(spl / 10)`.  Use `np.sum`, with the `axis=` directive if you need to sum a NumPy array rather than generic sum.